### DISCUSSION TITLE
WiFi access data still exists after VS Code download

### DISCUSSION THREAD

**Discussion by 011V32** [rednblkx/HomeKey-ESP32#45] [Q&A] [answered]
Is it actually normal that the WiFi access data remains after downloading/flashing the firmware again with VS Code?
What else is left after downloading/flashing the FW?

**Comment by rednblkx** [comment, marked as answer]
The persistent data is stored in the NVS(Non Volatile Storage) partition which is not overwritten during flashing, not my choice, just just PlatformIO operates by default.

There are currently 5 partitions as defined by the [with_ota.csv](https://github.com/rednblkx/HomeKey-ESP32/blob/777b6e12b3f6d9de1acab336a0e3a7489cf706c4/with_ota.csv) out of which 1(spiffs) is currently unused, the one nvs partition, another one(otadata) stores data for OTA updating, and 2(app0 and app1) store the actual code of this project which is what is actually overwritten upon flashing, there's two since when you want to OTA update you can't just overwrite the current running software therefore you need to write it somewhere else and keep of track of what partition should be used at boot which is what otadata does.

A brief introduction on how you can do OTA updates is available at the bottom of the README.md and i have it on my backlog to have it moved to the wiki and improve it.
